### PR TITLE
Marked plugin as thread safe

### LINKED ISSUE #211: Maven warns that plugin not thread safe
When I execute mvn package -T8, the following warning will be displayed:

[WARNING] *****************************************************************
[WARNING] * Your build is requesting parallel execution, but project      *
[WARNING] * contains the following plugin(s) that are not marked as       *
[WARNING] * @threadSafe to support parallel building.                     *
[WARNING] * While this /may/ work fine, please look for plugin updates    *
[WARNING] * and/or request plugins be made thread-safe.                   *
[WARNING] * If reporting an issue, report it against the plugin in        *
[WARNING] * question, not against maven-core                              *
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked @threadSafe in XXX:
[WARNING] de.jflex:jflex-maven-plugin:1.6.1
[WARNING] *****************************************************************

### PR DESCRIPTION
**Pull request by dheid**
Updated to maven-plugin-api 3.0.5 and introduced Mojo annotations. Marked JFlexMojo as thread-safe. Closes #211

### CODE REVIEW

**Comment by sarowe** [comment]
This looks great, thanks David!  But as Gerwin mentioned over on #211, there are known non-thread-safe things that need to be dealt with, e.g. converting static options to instance members, before we can declare the plugin thread-safe.

**Comment by dheid** [comment]
Is it possible to add at least the annotations and mark the plugin as non thread-safe until the conversion will be done? I think, the annotations are a useful improvement in general.

**Comment by lsf37** [comment]
I think the annotations are a good idea, happy to have them, if we mark it as non thread-safe for now. @sarowe do you agree?

**Comment by sarowe** [comment]
> I think the annotations are a good idea, happy to have them, if we mark it as non thread-safe for now. @sarowe do you agree?

+1

**Comment by dheid** [comment]
Nice!!! Thank you!

**Comment by lsf37** [comment]
Now merged and marked as not thread safe (for now) as discussed in 104b2197468039d132578b183ed44ed0d8ede503

Thanks for the contribution!

File: jflex-maven-plugin/pom.xml [hunk 1]
```diff
@@ -44,7 +44,7 @@
     <url>https://github.com/jflex-de/jflex/issues</url>
   </issueManagement>
   <prerequisites>
-    <maven>3.0.1</maven>
+    <maven>${maven.version}</maven>
```
Comment (regisd) [inline jflex-maven-plugin/pom.xml]: We need a very recent version of Maven to compile JFlex due to a bug in maven-shade-plugin.

Are you sure we actually need the same to execute the plugin? It sounds to me that 3.0 should be enough.

**Comment by regisd** [comment]
Thanks for updating the plugin to use Java annotations rather than invalid javadoc tags.

File: jflex-maven-plugin/pom.xml (hunk 1 above)
Comment (dheid) [reply jflex-maven-plugin/pom.xml]: Maven version 3 should really be enough. The annotation were introduced  back then.

**Comment by dheid** [comment]
You are welcome!

**Pull request merged**